dpnp/backend/extensions/lapack/common_helpers.hpp

```diff
@@ -63,4 +65,119 @@ inline bool check_zeros_shape(int ndim, const py::ssize_t *shape)
     }
     return src_nelems == 0;
 }
+
+// Allocate the memory for the pivot indices
+inline std::int64_t *alloc_ipiv(const std::int64_t n, sycl::queue &exec_q)
+{
+    std::int64_t *ipiv = nullptr;
+
+    try {
+        ipiv = sycl::malloc_device<std::int64_t>(n, exec_q);
+        if (!ipiv) {
+            throw std::runtime_error("Device allocation for ipiv failed");
+        }
+    } catch (sycl::exception const &e) {
+        if (ipiv != nullptr)
+            sycl::free(ipiv, exec_q);
+        throw std::runtime_error(
+            std::string(
+                "Unexpected SYCL exception caught during ipiv allocation: ") +
+            e.what());
+    }
+
+    return ipiv;
+}
+
+// Allocate the total memory for the total pivot indices with proper alignment
+// for batch implementations
+template <typename T>
+inline std::int64_t *alloc_ipiv_batch(const std::int64_t n,
+                                      std::int64_t n_linear_streams,
+                                      sycl::queue &exec_q)
+{
+    // Get padding size to ensure memory allocations are aligned to 256 bytes
+    // for better performance
+    const std::int64_t padding = 256 / sizeof(T);
+
+    // Calculate the total size needed for the pivot indices array for all
+    // linear streams with proper alignment
+    size_t alloc_ipiv_size = round_up_mult(n_linear_streams * n, padding);
+
+    std::int64_t *ipiv = nullptr;
+
+    // Allocate memory for the total pivot indices array
+    try {
+        ipiv = sycl::malloc_device<std::int64_t>(alloc_ipiv_size, exec_q);
+        if (!ipiv)
+            throw std::runtime_error("Device allocation for ipiv failed");
+    } catch (sycl::exception const &e) {
+        throw std::runtime_error(
+            std::string(
+                "Unexpected SYCL exception caught during ipiv allocation: ") +
+            e.what());
+    }
+
+    return ipiv;
```

Comment:
Is it the same like:
```suggestion
    return alloc_ipiv(alloc_ipiv_size, exec_q);
```
The same is applicable to `alloc_scratchpad_batch` also.

Reply:
That is a good point!
I will do the same for scratchpad. Thank you